### PR TITLE
feat: add inputs generator

### PR DESCRIPTION
**Pull request by kedevked**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ng-primitives/ng-primitives/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What does this PR implement/fix?

- Add a generator for an input component
- Add a generator for an input-form-field component

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/ng-primitives/schematics/input/schema.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright © 2024 Angular Primitives.
+ * https://github.com/ng-primitives/ng-primitives
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+export interface ButtonGeneratorSchema {
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/input/schema.d.ts]:
```suggestion
export interface InputGeneratorSchema {
```

File: packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div ngpFormField>
+  <label ngpLabel>label</label>
+  <p ngpDescription>description</p>
+  <input ngpInput placeholder="placeholder" />
+</div>
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template]: I'm not sure how useful an `input-form-field` component would be in general to have in an application. There are too many things someone may want to customize for each form field for it to be abstracted into one component.

For example, if I use this generator to create a component, and I go to build a form using it, e.g:

```html
<form>

<!-- Email -->
<app-input-form-field></app-input-form-field>

<!-- Name -->
<app-input-form-field></app-input-form-field>

</form>
```

How would I define the label text, the placeholder text, the input field type attribute, the description text, any errors, connect Angular forms etc.. There are too many complexities to make this useful. 

I think a better approach is to continue the way you have been with button and input, where we generate a component for each individual primitive, e.g. form-field, label, description etc.. That way they can be pieced together however they need, but it still speeds up the setup for someone adding them to their application 😊

File: packages/ng-primitives/schematics/input/files/__name__.component.__style__.template [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+[ngpInput] {
+}
+
+[ngpInput][data-focus='true'] {
+}
+
+[ngpInput]::placeholder {
+}
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/input/files/__name__.component.__style__.template]:
```suggestion
:host {
}

:host[data-focus='true'] {
}

:host::placeholder {
}
```

File: packages/ng-primitives/schematics/input/files/__name__.component.ts.template [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    [ngpInput] {
+    }
+
+    [ngpInput][data-focus='true'] {
+    }
+
+    [ngpInput]::placeholder {
+    }
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/input/files/__name__.component.ts.template]:
```suggestion
    :host {
    }

    :host[data-focus='true'] {
    }

    :host::placeholder {
    }
```

File: packages/ng-primitives/schematics/input/generator.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * https://github.com/ng-primitives/ng-primitives
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { formatFiles, generateFiles, names, Tree } from '@nx/devkit';
+import * as path from 'path';
+import { ButtonGeneratorSchema } from './schema';
+
+export async function inputGenerator(tree: Tree, options: ButtonGeneratorSchema) {
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/input/generator.ts]:
```suggestion
export async function inputGenerator(tree: Tree, options: InputGeneratorSchema) {
```

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Thank you very much for the contribution! Looks great, a few small suggestions.

File: packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template (hunk 2 above)
Comment (kedevked) [reply packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template]: I think the `input-form-field` generator is useless. What we need rather is a `form-field` generator right ?

Comment (ashley-hunter) [reply packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template]: Exactly 👍

Comment (kedevked) [reply packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template]: Done

Comment (ashley-hunter) [reply packages/ng-primitives/schematics/input-form-field/files/__name__.component.html.template]: Thank you!

**Review by ashley-hunter — COMMENTED** [review]
Great work! Thank you!

**Pull request merged**